AbletonMCP_Remote_Script/handlers/midi.py

```
@@ -0,0 +1,411 @@
+        }
+    except Exception as e:
+        if ctrl:
+            ctrl.log_message("Error clearing clip notes: " + str(e))
+        raise
+
+
+def quantize_clip_notes(song, track_index, clip_index, grid_size, ctrl=None):
+    """Quantize MIDI notes in a clip to a grid."""
+    try:
+        if grid_size <= 0:
+            raise ValueError("grid_size must be greater than 0")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)